src/components/DataTable.js

```diff
@@ -54,13 +107,14 @@ export default class DataTable extends React.Component {
 
   render() {
     const props = this.props;
+    let values = this.sortRows(props.values);
     const length = props.values.length;
     const limit = props.limit;
     const page = this.state.page;
     const start = page * limit;
     const end = start + limit;
-    const values = props.values.slice(start, end);
-    const columns = Object.keys(values[0] || {});
+    values = values.slice(start, end);
+    const columns = props.headers;
     const standardColumnWidth = 100;
     const wide = (columns.length * standardColumnWidth < window.innerWidth);
```

Comment:
We have `wide` in the initial state, but it doesn't seem to be used anywhere -- should this calc (and `standardColumnWidth`) be moved to the constructor?